google-cloud-bigtable-stats/src/main/java/com/google/cloud/bigtable/stats/BuiltinMetricsRecorder.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.stats;
+
+import com.google.api.core.InternalApi;
+import com.google.api.gax.tracing.ApiTracerFactory.OperationType;
+import com.google.api.gax.tracing.SpanName;
+import io.opencensus.stats.MeasureMap;
+import io.opencensus.stats.StatsRecorder;
+import io.opencensus.tags.TagContext;
+import io.opencensus.tags.TagContextBuilder;
+import io.opencensus.tags.TagKey;
+import io.opencensus.tags.TagValue;
+import io.opencensus.tags.Tagger;
+import io.opencensus.tags.Tags;
+import java.util.Map;
+
+/** Add built-in metrics to the measure map * */
+@InternalApi("For internal use only")
+public class BuiltinMetricsRecorder {
+
+  private final OperationType operationType;
+
+  private final Tagger tagger;
+  private final StatsRecorder statsRecorder;
+  private final TagContext parentContext;
+  private final SpanName spanName;
+  private final Map<String, String> statsAttributes;
+
+  private MeasureMap attemptLevelNoStreaming;
+  private MeasureMap attemptLevelWithStreaming;
+  private MeasureMap operationLevelNoStreaming;
+  private MeasureMap operationLevelWithStreaming;
```

Comment:
the WithStreamin and NoStreaming is very hard to follow. Do non-streaming operations not have latencies?